chore(dart/transform): Allow pub (build|serve) to specify mode

kegluneq opened the pull request:
Update pubbuild.js & pubserve.js to allow the caller to provide a `mode`
value.

Create gulp targets that pass `--mode=ngstatic` to `pub serve` in order to activate Dart
transformers.

Update the benchmark pubspec.yaml to use Angular 2 transformers.

yjbanov commented:
What if I want to run dart2js in 'release' mode and ng2 transformers in 'ngstatic' simultaneously?

kegluneq commented:
You currently can't without a 1-line code change [here](https://github.com/angular/angular/blob/master/modules/angular2/src/transform/transformer.dart#L25). My feeling is that there are few enough of us who are interested in that scenario at this point.

kegluneq commented:
I should have been more explicit about this upfront, but the `--mode` parameter is just meant as a temporary measure. I don't think the transformers are mature enough to inflict them on everyone by default yet :)

Commit:
fix(dart/transform): Do not import reflection_capabilities.ng_deps.dart

Run `ReflectionRemover` prior to `DirectiveProcessor` and
`DirectiveLinker`, otherwise the generated code will
maintain the transitive import to `dart:mirrors`.

Closes #1246

Commit:
refactor(dart/transform): Update ReflectionRemover to use AssetReader

To keep phases consistent, use `AssetReader` and `AssetId` to provide
information to `ReflectionRemover`.

Commit:
feat(dart/transform): Allow multiple transformer entry points

Allow the user to specify multiple entry points to an app.

Commit:
fix(dart/transform): Gracefully handle log calls before init

- Lazily create and use a logger that prints instead of `throw`ing.
- Use this logger in unit tests.

Commit:
feat(dart/transform): Update transformer to handle cases in benchmarks

* Add a few methods to the Dart html adapter
* Handle some reserved js words
* Sanitize property names with '$' in them

Commit:
feat(dart/transform): Run the transformer on benchmarks

Remove explicit generation of reflection information in benchmark code
and generate it with the transformer.
